packages/web/app/features/ProgramRegistry/PatientProgramRegistryUpdateFormModal.jsx

```diff
@@ -0,0 +1,337 @@
+import React from 'react';
+import * as yup from 'yup';
+import styled from 'styled-components';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { Add } from '@material-ui/icons';
+import MuiDivider from '@material-ui/core/Divider';
+import {
+  AutocompleteField,
+  Button,
+  DateDisplay,
+  DateField,
+  Field,
+  Form,
+  Heading5,
+  Modal,
+  ModalFormActionRow,
+  TextButton,
+  TextField,
+  TranslatedText,
+} from '../../components';
+import { useApi, useSuggester } from '../../api';
+import { optionalForeignKey } from '../../utils/validation';
+import { PANE_SECTION_IDS } from '../../components/PatientInfoPane/paneSections';
+import { Colors, FORM_TYPES } from '../../constants';
+import { ProgramRegistryConditionCategoryField } from './ProgramRegistryConditionCategoryField';
+import { FormTable } from './FormTable';
+import { usePatientProgramRegistryConditionsQuery } from '../../api/queries';
+import { ProgramRegistryConditionField } from './ProgramRegistryConditionField';
+
+const StyledFormTable = styled(FormTable)`
+  table tr td {
+    border: none;
+  }
+`;
+
+const Divider = styled(MuiDivider)`
+  margin: 10px 0;
+`;
+
+const StyledTextField = styled(TextField)`
+  .Mui-disabled {
+    background-color: #f3f5f7;
+  }
+`;
+
+const StyledAutocompleteField = styled(AutocompleteField)`
+  width: 300px;
+`;
+
+const AddButton = styled(Button)`
+  position: absolute;
+  padding-left: 10px;
+
+  .MuiSvgIcon-root,
+  .MuiButton-startIcon {
+    margin-right: 0;
+  }
+`;
+
+const ViewHistoryButton = styled(TextButton)`
+  color: ${Colors.primary};
+  font-size: 0.6875rem;
+  text-decoration: underline;
+  padding: 0.3rem 1rem;
+  text-transform: none;
+  &:hover {
+    text-decoration: underline;
+  }
+`;
+
+const useUpdateProgramRegistryMutation = patientId => {
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  return useMutation(
+    data => {
+      return api.post(`patient/${patientId}/programRegistration`, data);
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries([`infoPaneListItem-${PANE_SECTION_IDS.PROGRAM_REGISTRY}`]);
+        queryClient.invalidateQueries(['patient', patientId]);
+      },
+    },
+  );
+};
+
+const getGroupedData = rows => {
+  const groupMapping = {
+    confirmedSection: [
+      'suspected',
+      'underInvestigation',
+      'confirmed',
+      'unknown',
+      'inRemission',
+      'notApplicable',
+    ],
+    resolvedSection: ['disproven', 'resolved'],
+    recordedInErrorSection: ['recordedInError'],
+  };
+
+  // Initialize result object
+  const groupedData = { confirmedSection: [{}], resolvedSection: [], recordedInErrorSection: [] };
```

Comment:
The data needs to be grouped so that the table can be laid out in sections. The sections have these different statuses but also determine styles and the placement and interaction with the AddFieldButton